app/models/pawn.rb

```diff
@@ -1,2 +1,54 @@
 class Pawn < Piece
+  def move_legal?(to_row, to_col)
+    return true if single_advance?(to_row, to_col)
+    return true if double_advance?(to_row, to_col)
+    return true if capture_move?(to_row, to_col)
+    return true if en_passant?(to_row, to_col) # need to send signal to capture logic for en passant
+    false
+  end
+
+  private
+
+  def single_advance?(to_row, to_col)
+    to_row == row + forward_one && to_col == col
+  end
+
+  def double_advance?(to_row, to_col)
+    to_row == row + forward_two && to_col == col && moves.empty?
+  end
+
+  def capture_move?(to_row, to_col)
+    to_row == row + forward_one && (to_col - col).abs == 1 && enemy_occupied?(to_row, to_col)
+  end
+
+  def en_passant?(to_row, to_col)
+    return false unless to_row == row + forward_one && (to_col - col).abs == 1
+    enemy = enemy_occupied?(to_row + back_one, to_col)
+    enemy.class == Pawn && enemy.en_passant_vuln?
```

Comment:
`enemy_occupied?` can return `nil`, so this should probably be prefixed with `enemy &&`